cfgrammar/src/lib/yacc/ast.rs

```diff
@@ -114,6 +114,7 @@ pub struct GrammarAST {
     // map from a rule name to indexes into prods
     pub rules: IndexMap<String, Rule>,
     pub prods: Vec<Production>,
+    pub token_directives: IndexSet<String>,
```

Comment:
I guess an alternate implementation here makes this a `Vec<Tidx<StorageT>>`, so it would take an index into the `tokens` field instead of cloning the string?

Reply:
I went ahead and did that in [6b0ea90](https://github.com/softdevteam/grmtools/pull/558/commits/6b0ea909c7a79a7d191c34e99fb8f315dc41fb4b)

The if statement isn't quite as nice, but I felt since the field was added `pub` like the rest, it was a better choice
than cloning the string. Perhaps it will look nicer with a `let chain` once those are stabilized?